### DISCUSSION TITLE
Provide a method to quickly distinguish published entries from scheduled entries

### DISCUSSION THREAD

**Discussion by murtaugh** [ExpressionEngine/ExpressionEngine#2912] [Feature Requests]
Because of the way statuses interact with entry dates, looking at a list of entries means you need to scan the column of dates before you can tell which entries are open and published and which entries are open but unpublished.

One thing to do could be to bring Status and Date closer together in the table, making it clearer they interact in an important way. Another option would be to append "pending" to any dates in the future.

<img width="1061" alt="Screen Shot 2021-04-26 at 5 10 39 PM" src="https://user-images.githubusercontent.com/308752/116151442-98a0da00-a6b2-11eb-8004-801bd01e36a7.png">

It would also be great if the Date filter provided a dedicated option for seeing all entries with a date in the future, which I described in #513.

**Comment by PaulBaileyAtChan** [comment]
In the general case, an entry can be in one of three states with respect to the date: "pending", "published", and "expired", and it would be nice to be able to see all three at a glance, and to filter on each, as suggested above. The terminology here does need to be carefully chosen, because it needs to interact with the entry statuses in a way that's not confusing.

Something that's tricky is that "pending" is determined relative to the Date field, but "expired" is determined relative to the Expiration Date field, and that's not displayed by default.

**Comment by amphibian** [comment]
One issue with applying descriptors like "(pending)" to future-dated entries is that it makes the assumption that an entry date is only serving as a _publish_ date. What if your channel holds events, in which case that entry is actually "published" but just with a future date, not "pending" at all?

**Comment by murtaugh** [comment]
Fair question. One of my long-term clients is a church, and half the entries they publish are events.

We use the entry date field for the event date, and they have noted that the current situation both makes it hard for them to filter out events that have already happened, and to distinguish future events at a glance.

I suggested the label "pending" instead of "published" for this very reason, to allow for some ambiguity. In the case of an events channel, the label "pending" has a different connotation, but one I'd feel comfortable educating my clients on.

**Comment by PaulBaileyAtChan** [comment]
The terminology here is secondary to the functionality. The ability to see at a glance whether *now* is before the first date, between first and last dates, or after the last date, is what's important. Could be done with icons, or colours, or whatever.

*But*, I will say that the idea of the first date as the *publication* date, and the second date as the *expiration* date, is already built in to EE. The Date tab describes the first date as "Date of publication for this entry", and the second date as "Date this entry should expire". Maybe those should be generalised, or customisable, but at the moment, thinking of the dates as publication/expiration is totally consistent with how EE sees them.

As an aside — of course the date fields can be used in other ways, but my experience is that it gets confusing to have the date fields mean one thing in one channel, and another thing in another channel. Better IMO to leave the built-in date fields as publish/expiration dates, and use custom fields for — for example — event dates. You might need dates for both things. For example, if you used an entry to announce an event, you might want that *announcement* to only appear after a certain date (publication date), or up to a certain date (expiration date) with the dates of the event itself defined separately, using custom fields.

So, IMO, seeing the built-in date fields as publication/expiration is *how EE already sees them*, and something to hold on to.

**Comment by murtaugh** [comment]
> The Date tab describes the first date as "Date of publication for this entry"

I hadn't noticed that, yeah; generalizing that description sounds like a good idea. I think "date this entry should expire" is already fairly general.

We might be getting off topic slightly — even if you accept that EE sees the entry date as when the post goes live, it's still hard to tell at a glance which entries have hit their entry date and which haven't (and no way to quickly filter for them, as described in #513).

**Comment by avnerw** [comment]
> The ability to see at a glance whether _now_ is before the first date, between first and last dates, or after the last date, is what's important. Could be done with icons, or colours, or whatever.

This is something I am so used to in the current CMS I'm using for the past decade, that I was suprised it is not possible in EE. It would be a great enhancement if at the minimum, a colored icon can appear next to the date and "expired" date like this. 

![Screen Shot 2021-07-07 at 1 44 12 PM](https://user-images.githubusercontent.com/17215743/124747265-6caf9980-df2a-11eb-8036-de84e9582207.png)

**Comment by intoeetive** [comment]
@avnerw I do like the UI suggestion!

Can you add some sort of legend, which icon is representing what?

(also, which is the other CMS you're getting inspired with here? would like to have a look)

**Comment by avnerw** [comment]
@intoeetive I am not exactly sure if there _is_ an automatic relationship in EE between the Status and the Publish date/Expiry date of an entry, so my example of icons above might not be entirely clear. That begs the question why an expired entry might still have an "Open" status. But here is what I thought when I created that.  

![Screen Shot 2021-07-07 at 2 11 29 PM](https://user-images.githubusercontent.com/17215743/124749818-653dbf80-df2d-11eb-83d6-3d105863e2cc.png)

**Comment by intoeetive** [comment]
`status` is the property on `ChannelEntry` model, it is not being updated to anything when dates are changing. Main reason is that you can use custom statuses. So `status` is not the only thing that determines whether entry is "active" or not - it's the combination of status and dates that does (and I'm not touching the aspect that you can override that...)

Thank you for the explanation, that looks more clear now!

**Comment by avnerw** [comment]
Ok, so I now understand the status issue. I was quite used to the status being directly linked to the publishing/expiry date schedule in Movable Type, which is the CMS I was referring to.  I guess the use of an icon would enhance the listing screen quite a bit. Not sure if one would even need to give an option to be able to change the color of the icons. I guess an entry with a _closed_ status can have a neutral icon that does not conflict with any of the others.

**Comment by intoeetive** [comment]
Another proposed solution is in https://github.com/ExpressionEngine/ExpressionEngine/pull/3469

It does not show icons everywhere of affect status display, but only adds an icon where special attention might be needed:
- to entry date column ONLY if entry date is in future
- to expiration date column ONLY if expiration is in the past

![image](https://github.com/ExpressionEngine/ExpressionEngine/assets/752126/dce7b0fd-b550-414b-83fa-de60dfadb438)

**Reply by jaredchesebro** [reply]
I like this idea.

There was a bit of talk about the correct terminology for an entry that has a future date and/or expiration date. I think proper terminology would be "scheduled" instead of "pending". Thinking that would help for any accessibility labels added to this feature.

Also, I wonder if appending the icon before the post title, rather than the dates would provide a little more clarity and make it easier to quickly scan for posts that are "scheduled".
